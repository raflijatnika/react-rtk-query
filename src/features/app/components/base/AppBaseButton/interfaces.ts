// Antd
import { ButtonProps } from 'antd'

// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

export interface IAppBaseButtonProps extends ButtonProps {
  backgroundColor?: APP_COLOR | APP_COLOR_LIGHT
  fontColor?: APP_COLOR | APP_COLOR_LIGHT
  width?: number
  height?: number
}
