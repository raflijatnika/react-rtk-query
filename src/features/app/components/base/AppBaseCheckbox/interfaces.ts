// Antd
import { CheckboxProps } from 'antd'

export type IAppBaseCheckboxProps = CheckboxProps
