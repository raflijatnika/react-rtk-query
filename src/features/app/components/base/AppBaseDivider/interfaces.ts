// Antd
import { DividerProps } from 'antd'

export type IAppBaseDividerProps = DividerProps
