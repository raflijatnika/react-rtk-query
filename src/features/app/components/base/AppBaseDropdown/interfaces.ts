// React
import { ReactElement } from 'react'

// Interfaces
import { DropdownProps, MenuProps } from 'antd'

export interface IDropdownProps {
  children?: ReactElement
  trigger?: DropdownProps['trigger']
  items: MenuProps['items']
}
