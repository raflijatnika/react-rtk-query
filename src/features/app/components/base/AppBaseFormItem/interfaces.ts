// Interfaces
import { FormItemProps } from 'antd'

// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

export interface IAppBaseFormItemProps extends FormItemProps {
  labelColor?: APP_COLOR | APP_COLOR_LIGHT
  labelSize?: number
}
