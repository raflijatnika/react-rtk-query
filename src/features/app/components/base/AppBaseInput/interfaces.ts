// Interfaces
import { InputProps } from 'antd'

// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

export interface IAppBaseInputProps extends InputProps {
  backgroundColor?: APP_COLOR | APP_COLOR_LIGHT
  height?: number
  width?: number
}
