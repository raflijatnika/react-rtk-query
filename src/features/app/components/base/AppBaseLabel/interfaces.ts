// Constant
import {
  APP_COLOR,
  APP_COLOR_LIGHT
} from '@/features/app/constant/app-style.constant'

export interface IAppBaseLabelProps {
  fontWeight?: number
  fontSize?: number
  isBold?: boolean
  fontColor?: APP_COLOR | APP_COLOR_LIGHT
}
