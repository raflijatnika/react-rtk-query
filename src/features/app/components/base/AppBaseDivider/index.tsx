// Styled Components
import styled from 'styled-components'

// Antd
import { Divider } from 'antd'

// Interfaces
import { IAppBaseDividerProps } from './interfaces'

const AppBaseDivider = styled(Divider).attrs(
  (props): IAppBaseDividerProps => props
)<IAppBaseDividerProps>`
  display: flex;
  align-items: center;
`

export { AppBaseDivider }
