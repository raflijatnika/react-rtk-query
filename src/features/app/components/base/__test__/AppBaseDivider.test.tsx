// Test
import { render } from '@testing-library/react'
import renderer, { ReactTestRenderer } from 'react-test-renderer'
import 'jest-styled-components'

// Components
import { AppBaseDivider } from '../AppBaseDivider'

// Interfaces
import { IAppBaseDividerProps } from '../AppBaseDivider/interfaces'

const rendererInstance = (props?: IAppBaseDividerProps): ReactTestRenderer =>
  renderer.create(<AppBaseDivider {...props} />)

it('AppBaseDivider should render', () => {
  render(<AppBaseDivider />)
})

it('AppBaseDivider should have snapshot', () => {
  const tree = rendererInstance().toJSON()

  expect(tree).toMatchSnapshot()
})

it('AppBaseDivider should have default style', () => {
  const tree = rendererInstance().toJSON()

  expect(tree).toHaveStyleRule('display', 'flex')
  expect(tree).toHaveStyleRule('align-items', 'center')
})
