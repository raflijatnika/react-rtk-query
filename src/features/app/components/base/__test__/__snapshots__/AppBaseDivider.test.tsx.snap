// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBaseDivider should have snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="ant-divider ant-divider-horizontal c0"
  role="separator"
  theme={Object {}}
/>
`;
